docs/reST/ref/rect.rst

```diff
@@ -348,6 +348,52 @@
       with the Rect. If no intersecting rectangles are found, an empty list is
       returned.
 
+      Not only Rects are valid arguments, but these are all valid calls:
+
+      .. code-block:: python
+          :linenos:
+    
+          Rect = pygame.Rect
+          r = Rect(0, 0, 10, 10)
+
+          list_of_rects = [Rect(1,1,1,1), Rect(2,2,2,2)]
+          indices0 = r.collidelistall(list_of_rects)
+    
+          list_of_lists = [[1,1,1,1], [2,2,2,2]]
+          indices1 = r.collidelistall(list_of_lists)
+    
+          list_of_tuples = [(1,1,1,1), (2,2,2,2)]
+          indices2 = r.collidelistall(list_of_tuples)
+    
+          list_of_double_tuples = [((1,1), (1,1)), ((2,2), (2,2))]
+          indices3 = r.collidelistall(list_of_double_tuples)
+    
+          class ObjectWithRectAttribute(object):
+    
+              def __init__(self, r):
+                  self.rect = r
+    
+          list_of_object_with_rect_attribute = [
+                                               ObjectWithRectAttribute(Rect(1,1,1,1)), 
+                                               ObjectWithRectAttribute(Rect(2,2,2,2))
+          ]
+          indices4 = r.collidelistall(list_of_object_with_rect_attribute)
+    
+          class ObjectWithCallableRectAttribute(object):
+    
+              def __init__(self, r):
+                  self._rect = r
+    
+              def rect(self):
+                  return self._rect
+    
+          list_of_object_with_callable_rect = [
+                                              ObjectWithCallableRectAttribute(Rect(1,1,1,1)), 
+                                              ObjectWithCallableRectAttribute(Rect(2,2,2,2))
+          ]
+          indices5 = r.collidelistall(list_of_object_with_callable_rect)
```

Comment:
minor nitpick, this codeblock could do better with `black` formatting. We use black throughout the codebase, so as to maintain consistency ;)

Reply:
your are right, reformatted example code with black